shiny/session/_session.py

```diff
@@ -404,6 +417,43 @@ async def _send_response(self, message: ClientMessageOther, value: object) -> No
 
     # This is called during __init__.
     def _create_message_handlers(self) -> dict[str, Callable[..., Awaitable[object]]]:
+        # TODO-future; Make sure these methods work within MockSession
+
+        async def outputRPC(
+            outputId: str,
+            handler: str,
+            msg: object,
+        ) -> object:
+            """
+            Used to handle request messages from an Output Renderer.
+
+            Typically, this is used when an Output Renderer needs to set an input value.
+            E.g. a @render.data_frame result with a `mode="edit"` parameter will use
+            this to set the value of the data frame when a cell is edited.
+            """
+            with session_context(self):
+                if not (outputId in self.output._outputs):
+                    raise RuntimeError(
+                        f"Received request message for an unknown Output Renderer with id `{outputId}`"
+                    )
+
+                output_info = self.output._outputs[outputId]
+                renderer = output_info.renderer
+                handler = f"_handle_{handler}"
+                if not (
+                    hasattr(renderer, handler) and callable(getattr(renderer, handler))
+                ):
+                    raise RuntimeError(
+                        f"Output Renderer with id `{outputId}` does not have method `{handler}` to handler request message"
+                    )
+                try:
+                    return await getattr(renderer, handler)(msg)
```

Comment:
Handler methods should manage their own session context, in a way that works with modules.